### PR TITLE
Clean up builtin Map and Set types

### PR DESCRIPTION
**Pull request by ttuegel**
- Finish breaking up the `Builtin` type so it is no longer used.
- Rename the `BuiltinMap_` and `BuiltinList_` patterns to `InternalMap_` and `InternalList_`.
- Test that simplification preserves sorts of `InternalMap` and `InternalList`.

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [x] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [x] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [x] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [x] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

File: kore/src/Kore/Builtin/AssociativeCommutative.hs [hunk 1]
```diff
@@ -130,18 +129,30 @@ import qualified Pretty
 @InternalAc@ struct inside a @Domain.Builtin.Builtin@ value.
```
Comment (ana-pantilie) [inline kore/src/Kore/Builtin/AssociativeCommutative.hs:129]: `Domain.Builtin.Builtin` doesn't exist anymore, right?

File: kore/src/Kore/Builtin/AssociativeCommutative.hs [hunk 2]
```diff
@@ -171,24 +182,27 @@ class
         -> Maybe (normalized (TermLike Concrete) (TermLike variable))
 
     simplifiedAttributeValue
-        :: Domain.Value normalized (TermLike variable) -> Attribute.Simplified
+        :: Value normalized (TermLike variable) -> Attribute.Simplified
 
-instance TermWrapper Domain.NormalizedMap where
-    {- | Render a 'NormalizedMap' as a Domain.Builtin.
+instance TermWrapper NormalizedMap where
+    {- | Render a 'NormalizedMap' as a Builtin.
```
Comment (ana-pantilie) [inline kore/src/Kore/Builtin/AssociativeCommutative.hs:188]: Should this be 
```suggestion
    {- | Render a 'NormalizedMap' as an 'InternalAc'.
```
?

File: kore/src/Kore/Builtin/AssociativeCommutative.hs [hunk 3]
```diff
@@ -233,30 +247,33 @@ instance TermWrapper Domain.NormalizedMap where
...
             , opaque = [patt]
             }
 
-    simplifiedAttributeValue = TermLike.simplifiedAttribute . Domain.getMapValue
+    simplifiedAttributeValue = TermLike.simplifiedAttribute . getMapValue
 
-instance TermWrapper Domain.NormalizedSet where
-    {- | Render a 'NormalizedSet' as a Domain.Builtin.
+instance TermWrapper NormalizedSet where
+    {- | Render a 'NormalizedSet' as a Builtin.
```
Comment (ana-pantilie) [inline kore/src/Kore/Builtin/AssociativeCommutative.hs:259]: Should this be
```suggestion
    {- | Render a 'NormalizedSet' as an 'InternalAc'.
```
?

File: kore/src/Kore/Internal/TermLike.hs [hunk 4]
```diff
@@ -57,7 +56,6 @@ module Kore.Internal.TermLike
     , mkBottom
     , mkInternalBytes
     , mkInternalBytes'
-    , mkBuiltin
     , mkInternalBool
     , mkInternalInt
     , mkInternalString
```
Comment (ana-pantilie) [inline kore/src/Kore/Internal/TermLike.hs:61]: I can't comment on the specific lines, but shouldn't the `mkBuiltin...` functions for collections be renamed to `mkInternal...` as well?

File: kore/src/Kore/Internal/InternalMap.hs [hunk 5]
```diff
@@ -0,0 +1,147 @@
+{-|
+Copyright : (c) Runtime Verification, 2018
```
Comment (andreiburdusa) [inline kore/src/Kore/Internal/InternalMap.hs:2]: 2021?

File: kore/src/Kore/Internal/InternalSet.hs [hunk 6]
```diff
@@ -0,0 +1,149 @@
+{- |
+Copyright   : (c) Runtime Verification, 2018
```
Comment (andreiburdusa) [inline kore/src/Kore/Internal/InternalSet.hs:2]: 2021?

File: kore/src/Kore/Step/Simplification/InternalMap.hs [hunk 7]
```diff
@@ -0,0 +1,54 @@
+{- |
+Copyright   : (c) Runtime Verification, 2019
```
Comment (andreiburdusa) [inline kore/src/Kore/Step/Simplification/InternalMap.hs:2]: 2021? :)

File: kore/src/Kore/Step/Simplification/InternalSet.hs [hunk 8]
```diff
@@ -0,0 +1,55 @@
+{- |
+Copyright   : (c) Runtime Verification, 2019
```
Comment (andreiburdusa) [inline kore/src/Kore/Step/Simplification/InternalSet.hs:2]: 2021? :)

**Review by andreiburdusa — APPROVED** [review]

**Pull request merged**